### PR TITLE
Fix creation of Koji Build Finder NVR list text file

### PR DESCRIPTION
**Pull request by dwalluck**

### CODE REVIEW

File: pig/src/main/java/org/jboss/pnc/bacon/pig/impl/nvr/NvrListGenerator.java [hunk 1]
```diff
@@ -42,24 +42,20 @@ public static boolean generateNvrList(String repoZipPath, String targetPath) {
...
+            log.error("Failed to write file {} to {}: {}", basename, outputDirectory, e.getMessage(), e);
             return false;
         }
 
-        File srcFile = new File(outputDirectory, nvrReport.getBaseFilename() + ".txt");
-        File destFile = new File(targetPath);
+        File destFile = new File(outputDirectory, basename);
+        boolean exists = destFile.exists();
 
-        try {
```
Comment (dwalluck) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/impl/nvr/NvrListGenerator.java:56]: File was already saved to location on line 47, then it tried to move the file on top of itself I think.

**Comment by rnc** [comment]
Is there a test for this?

**Comment by dwalluck** [comment]
> Could you make one please?

I've added a test. I did this by mocking `BuildFinderUtils.findBuilds()` (which normally requires networking) to return a builds list loaded from `src/test/resources/builds.json`.

**Comment by dwalluck** [comment]
retest this please

File: pig/src/test/java/org/jboss/pnc/bacon/pig/impl/nvr/NvrListGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class NvrListGeneratorTest {
+    @Test
+    void shouldGenerateNvrList(@TempDir Path tempDir) throws URISyntaxException, IOException {
+        try (MockedStatic<BuildFinderUtils> utilsMockedStatic = Mockito.mockStatic(BuildFinderUtils.class)) {
```
Comment (rnc) [inline pig/src/test/java/org/jboss/pnc/bacon/pig/impl/nvr/NvrListGeneratorTest.java:28]: Nice!

**Review by rnc — APPROVED** [review]
LGTM

**Comment by dwalluck** [comment]
@thescouser89 @rnc I am not sure why the codecov hasn't rerun. I added a test so the coverage should be increasing.

**Comment by dwalluck** [comment]
> @thescouser89 @rnc I am not sure why the codecov hasn't rerun. I added a test so the coverage should be increasing.

Worked now.

**Pull request merged**